lib/core/include/irods/authentication_plugin_framework.hpp

```diff
@@ -162,6 +162,10 @@ namespace irods::experimental::auth
     /// \since 4.3.0
     inline void authenticate_client(RcComm& _comm, const RodsEnvironment& _env, const json& _ctx)
     {
+        if (_comm.loggedIn) {
+            return;
+        }
+
```

Comment:
What's the reason for adding this? Did you notice something?

Reply:
I had made some more extensive changes which were not included in this PR. I think this change either fixed an issue related to those changes or was just here as a way to save a few cycles. I left it in a separate commit in case anybody objected to its inclusion.

Reply:
Seems fine to me.

This is about silently ignoring the call to `authenticate_client` if already logged in, correct? This is purely on the client side of things, right?

Reply:
I would not say that it is pure because this is supposed to serve a similar purpose to `clientLogin`. Servers could call it when redirecting as well.

I suppose it technically causes the first operation to be skipped, but the plugins we have now would not really be affected. See the control loop below.

Reply:
Got it.

Reply:
One more question - do you feel that check is necessary/good?

Reply:
It's an optimization - not necessary, but elides a (probably) unnecessary call into the plugin operation. Even if the `rcComm.loggedIn` flag is set erroneously, it would only get to call the start operation of the plugin below before returning and would do nothing.

Reply:
Good enough for me.